<a href="https://colab.research.google.com/github/fandri-indranata/data-science-2026/blob/main/Pertemuan2_FandriIndranata_230401010180ipynb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Nama: Fandri Indranata NIM: 230401010180 Kelas: IF401

In [1]:
import pandas as pd
import numpy as np

# Dataset Titanic
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

# Menampilkan 5 baris pertama
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [2]:
# Q1: Berapa jumlah baris dan kolom pada dataset Titanic? Kolom apa saja yang tersedia?
print("Jumlah Baris dan Kolom (Rows,Columns):",df.shape)
print("\nDaftar Kolom yang tersedia:")
print(df.columns.tolist())

Jumlah Baris dan Kolom (Rows,Columns): (891, 12)

Daftar Kolom yang tersedia:
['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


Pada dataset titanic ini memiliki 891 baris dengan 12 kolom, kolom-kolom ini terdiri dari 'PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin' dan 'Embarked'.

In [3]:
# Q2: Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%) per kolom?
missing_count = df.isnull().sum()
missing_percent = (df.isnull().sum() / len(df)) * 100
# Membuat DataFrame baru untuk menggabungkan jumlah dan persentase
missing_df = pd.DataFrame({
    'Jumlah_Missing': missing_count,
    'Persentase(%)': missing_percent.round(2)
})
# Hanya tampilkan kolom yang memiliki missing values
print("Kolom dengan Missing Values:")
print(missing_df[missing_df['Jumlah_Missing'] > 0])

Kolom dengan Missing Values:
          Jumlah_Missing  Persentase(%)
Age                  177          19.87
Cabin                687          77.10
Embarked               2           0.22


Terdapat 3 kolom yang memiliki missing values: Age (177 data / 19,87%), Cabin (687 data / 77,10%), dan Embarked (2 data / 0,22%)."

In [4]:
# Q3: Berapa persen penumpang yang selamat (Survived=1)? Gunakan operasi NumPy atau Pandas
print(f"Persentase penumpang selamat: {np.mean(df['Survived']) * 100:.2f}%")

Persentase penumpang selamat: 38.38%


Dengan menggunakan Numpy didapat hanya sekitar 38,38% penumpang yang selamat dari bencana tenggelamnya kapal Titanic.

In [5]:
# Q4: Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-rata usianya?
wanita_kelas1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]

# Menghitung jumlah dan rata-rata usia
jumlah_wanita_kelas1 = len(wanita_kelas1)
rata_usia = wanita_kelas1['Age'].mean()

print(f"Jumlah penumpang wanita kelas 1: {jumlah_wanita_kelas1} orang")
print(f"Rata-rata usia mereka: {rata_usia:.2f} tahun")

Jumlah penumpang wanita kelas 1: 94 orang
Rata-rata usia mereka: 34.61 tahun


Pada data titanic terdapat 94 penumpang wanita di kelas 1. Rata-rata usia mereka adalah sekitar 34,61 tahun

In [6]:
# Q5: Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas penumpang. Kelas mana yang paling tinggi?
tingkat_selamat_kelas = df.groupby('Pclass')['Survived'].mean() * 100

print("Tingkat Keselamatan per Kelas Penumpang (%):")
print(tingkat_selamat_kelas.round(2))

# Mengetahui kelas mana yang paling tinggi
kelas_tertinggi = tingkat_selamat_kelas.idxmax()
print(f"\nKelas dengan tingkat keselamatan tertinggi: Kelas {kelas_tertinggi}")

Tingkat Keselamatan per Kelas Penumpang (%):
Pclass
1    62.96
2    47.28
3    24.24
Name: Survived, dtype: float64

Kelas dengan tingkat keselamatan tertinggi: Kelas 1


Kelas 1 memiliki tingkat keselamatan tertinggi (sekitar 62,96%), diikuti Kelas 2 (47,28%), dan Kelas 3 (24,24%). Ini menunjukkan bahwa penumpang di kelas utama (kaya) memiliki peluang selamat yang jauh lebih besar.

In [7]:
# Q6: Analisis tambahan - Tingkat keselamatan berdasarkan Jenis Kelamin
tingkat_selamat_gender = df.groupby('Sex')['Survived'].mean() * 100

print("Tingkat Keselamatan berdasarkan Jenis Kelamin (%):")
print(tingkat_selamat_gender.round(2))

Tingkat Keselamatan berdasarkan Jenis Kelamin (%):
Sex
female    74.20
male      18.89
Name: Survived, dtype: float64


Ada dua hal menarik yang dapat dapat disimpulkan dari data titanic ini dimana, pertama semakin elit atau mewah kelas kapal semakin tinggi tingkat keselamatan, Kelas 1 memiliki tingkat keselamatan tertinggi (sekitar 62,96%), diikuti Kelas 2 (47,28%), dan Kelas 3 (24,24%). Ini menunjukkan bahwa penumpang di kelas utama (kaya) memiliki peluang selamat yang jauh lebih besar. Selanjutnya jenis kelamin juga terdapat gap dimana, penumpang wanita memiliki tingkat keselamatan sekitar 74,20%, sementara penumpang pria hanya 18,89%.

Kesimpulan:
1. url yang menggunakan https://github.com/datasciencedojo/datasets/blob/master/titanic.csv tidak dapat diproses, untuk dapat diproses gunakan url https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv
2. Dapat diinterpretasikan bahwa penumpang yang selamat di kelas 1 lebih besar dari 1/2 keseluruhan penumpang, pada kelas 2 lebih kecil dari 1/2 keseluruhan penumpang, dan pada kelas 3 hanya 1/4 dari keseluruhan penumpang yang selamat. Sedangkan berdasarkan jenis kelamin, pada wanita angka keselamatan mencapai lebih besar dari 3/4 keseluruhan penumpang dan pada pria lebih kecil dari 1/4 keseluruhan penumpang yang selamat.